### ISSUE TITLE
[DRAFT] CodeChecker invocation for single compilation entry with customizable behaviour

### ISSUE THREAD

**Issue by vodorok** [Ericsson/codechecker#4176]
The main difference compared to the standard analyze command would be:
* A direct compile command entry, (as a string)  could be specified in place of the compilation database. The following example demonstrates the usage.
```
CodeChecker analyze ...other flags....  '{"directory": "/proj/src", "command": "clang++ src.cpp", "file": "src.cpp" }'  ....other flags....
```
Because only a single file is being analyzed, we should only generate one output `.plist` file. 
* The path to this output file could be directly specified with the -o option. 
* Alternatively, if a folder is given (so we are talking about a report folder.) a shadow file tree could be created under this report folder, that would match the source directory structure. The .plist file would be created at the same location but inside the shadow file tree.
(kinda the same as the `.ast` files under the CTU folder.)

We should consider that there might be multiple analyzers. 
* We could create output files that prefix the output file name with an analyzer name (ugly).
* Create a "custom" plsit, containing the output of all analyzers, so we could identify the analyzer that produced the diagnostic message.

In this mode, there should not be any metadata or shared files created, instead a separate command should generate these, or compute them during store and parse.

**Comment by dkrupp** [comment]
Maybe the command line could be simply like:

CodeChecker analyze -b "<build command>" -o results.plist

So if the output is a single plist file, then all analysis output will be reported into this single file. No other file (like metadata.json etc) would be created to not cause conflict in the report directory.

Another suggestions:
CodeChecker analyze --generate_metadata -o metadata.json would simply create a metadata json with the given configuration

**Comment by dkrupp** [comment]
We had a discussion with @bruntib and it might be also good enough to concurrently call the existing `CodeChecker analyze --file` command and place a lock on concurrently written files in the report directory:

-`metadata.json`
-`compiler_info.json`
-`extdefmap.txt`
